owsi-core/owsi-core-components/owsi-core-component-jpa-more/src/main/java/fr/openwide/core/jpa/more/business/history/model/bean/AbstractHistoryLogObjectsBean.java

```diff
@@ -0,0 +1,45 @@
+package fr.openwide.core.jpa.more.business.history.model.bean;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import fr.openwide.core.commons.util.CloneUtils;
+
+public abstract class AbstractHistoryLogObjectsBean<T> {
+	
+	private T mainObject;
+	
+	private Object[] secondaryObjects;
+
+	protected AbstractHistoryLogObjectsBean(T mainObject, Object... secondaryObjects) {
+		this.mainObject = mainObject;
+		this.secondaryObjects = CloneUtils.clone(secondaryObjects);
+	}
+	
+	public T getMainObject() {
+		return mainObject;
+	}
+	
+	public Object[] getSecondaryObjects() {
+		return CloneUtils.clone(secondaryObjects);
+	}
+	
+	@Override
+	public boolean equals(Object obj) {
+		if (obj instanceof AbstractHistoryLogObjectsBean) {
+			AbstractHistoryLogObjectsBean<?> other = (AbstractHistoryLogObjectsBean<?>) obj;
+			return new EqualsBuilder()
+					.append(mainObject, other.mainObject)
```

Comment:
Maybe we should add a comment here (or on the method description): we only compare the main object. This is because, in a given database transaction, the secondary objects should be the same whenever the main object is the same. The secondary objects are only useful for history.